### ISSUE TITLE
Is there a way to enable or configure a number of simultaneous yt-dlp downloads?

### ISSUE THREAD

**Issue by Ben64** [alexta69/metube#358]
As it is, MeTube is the best thing I've found for downloading live streams from Twitch. However, if one stream is being downloaded, nothing else can be at the same time, and the extra streams gets completely missed. Allowing concurrent downloads that is configurable would be excellent for my use case.

**Comment by PikuZheng** [comment]
are you meaning #314

**Comment by Ben64** [comment]
> are you meaning #314

No, multiple threads is nice but not what I'm referring to. If I put two currently live streams from twitch into MeTube, only the first one will download, and the second one is stuck in queue waiting for the first to finish. This results in the second stream being potentially lost in time forever.

I'd like a separate instance of yt-dlp to be launched if configured to do so

**Comment by PikuZheng** [comment]
well... maybe the easist way is deploy multi cotainer.

**Comment by Ben64** [comment]
Wouldn't the easiest way be to just add this feature to the application?

**Comment by rrediske** [comment]
> Wouldn't the easiest way be to just add this feature to the application?

Nope. The easiest way by far would be to deploy a second container, especially these days with docker compose and such, we're talking maybe 5 minutes? I can't imagine rewriting metube for simultaneous downloads would take less than 5 minutes.

But in the end, it depends on how badly you want it. If it's not worth 5 minutes of your time, you probably don't want it as badly as you think you do.

**Comment by Ben64** [comment]
> > Wouldn't the easiest way be to just add this feature to the application?
> 
> Nope. The easiest way by far would be to deploy a second container, especially these days with docker compose and such, we're talking maybe 5 minutes? I can't imagine rewriting metube for simultaneous downloads would take less than 5 minutes.
> 
> But in the end, it depends on how badly you want it. If it's not worth 5 minutes of your time, you probably don't want it as badly as you think you do.

Having a queue that allows multiple concurrent downloads would be so much more convenient than adding more containers every time I want to download a stream

**Comment by tadeusz-niemiec** [comment]
Would be great to have that feature, in my case, I often need to download big playlists from YouTube to prepare for DJ sets. Spawning multiple docker containers wouldn't do much, as it would require me to manually split the playlist into chunks and distribute them between instances. Would appreciate it if you could consider adding it.
Thanks

**Comment by rdiaz738** [comment]
@tadeusz-niemiec @Ben64 not sure if this is what you guys are looking but made this:
https://hub.docker.com/r/notataco/metube

**Comment by TangentFoxy** [comment]
Consider renaming this to "[Feature Request] Multiple simulataneous YouTube downloads" or something similar? I made an issue that got confused with this one because there are two stages to downloading using this tool, and I was talking about downloads *from MeTube* rather than from YouTube.

**Comment by rdiaz738** [comment]
> > [@tadeusz-niemiec](https://github.com/tadeusz-niemiec) [@Ben64](https://github.com/Ben64) not sure if this is what you guys are looking but made this: https://hub.docker.com/r/notataco/metube
> 
> [@rdiaz738](https://github.com/rdiaz738) This works great, would you be able to make a pull request to add the concurrency feature to this?

Just did, hopefully it is merged

**Comment by alexta69** [comment]
Done in #602
